.github/workflows/build-proto-image.yml

```
@@ -3,13 +3,99 @@
 name: Proto Build Workflow
 
 on:
-  workflow_dispatch:
+  push:
+    branches:
+      - "main"
+    tags:
+      # semver, e.g. 1.2.0 (does not match 0.1.2)
+      - "[1-9]+.[0-9]+.[0-9]+"
+      # semver with prerelease info, e.g. 1.0.2-beta.1 or 1.2.3-rc.10
+      - "[1-9]+.[0-9]+.[0-9]+-[a-z]+.[0-9]+"
+      # do not match prerelease starting w/ 0, e.g. 1.0.2-beta.0 or 1.2.3-rc.01
+      - "![1-9]+.[0-9]+.[0-9]+-[a-z]+.[0]*"
+      # semver with date info, e.g. 1.0.2-20221125
+      - "[1-9]+.[0-9]+.[0-9]+-[0-9]+"
+      # do not match date starting w/ 0, e.g. 1.0.2-01232023
+      - "![1-9]+.[0-9]+.[0-9]+-[0]*"
   pull_request:
```

Add the following to not trigger on changes to README, etc on PRs

```suggestion
  pull_request:
      paths-ignore:
      # do not run if only markdown or other administrative files have been updated
         - "*.md"
         - "CODEOWNERS"
         - "LICENSE"
```